beacon_node/beacon_chain/src/data_availability_checker.rs

```diff
@@ -248,18 +257,14 @@ impl<T: BeaconChainTypes> DataAvailabilityChecker<T> {
     /// This method is used if the EL already has the blobs and returns them via the `getBlobsV2`
     /// engine method.
     /// More details in [fetch_blobs.rs](https://github.com/sigp/lighthouse/blob/44f8add41ea2252769bb967864af95b3c13af8ca/beacon_node/beacon_chain/src/fetch_blobs.rs).
-    pub fn put_engine_data_columns(
+    pub fn put_engine_data_columns<O: ObservationStrategy>(
```

Comment:
This could also potentially be merged with `put_gossip_data_columns`?

Reply:
Yeah good catch - since we now do verification on EL blobs, this function is no longer necessary (docs above is also outdated). 

With this change we do delay our import block time for a bit though and should make sure our gossip verification is fast (rn it's quite slow on large blob count) -  this just reminded me that skipping verification was an intentional decision made when this was originally implemented - i think if we go down this route, there might be some parts that we should optimise in the verify method, e.g. verifying the block header, signature etc - potentially we could just verify the kzg proof? cc @michaelsproul

Reply:
Addressed in 966dadae6f00f0e363953d42d6861e25388c7a2b

I think it may be worth taking out gossip verification though, but will have to revive this function (to accept kzg verified columns instead of gossip columns) - let me know what you think.

Reply:
I think the main reason we do gossip verification is because we propagate the columns on gossip so we want to make sure its valid.
We can avoid the proposer signature validation which I would guess is the other factor which takes some significant amount time in gossip verification (other than kzg validation)